core/metrics/exporter.go

```diff
@@ -0,0 +1,235 @@
+	coreStatus         = "core_status" */
+)
+
+var (
+	tenantsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      tenantsTotal,
+			Help:      "How many Tenants exist",
+		})
+
+	agentsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      agentsTotal,
+			Help:      "How many SHIELD Agents have been registered",
+		})
+
+	targetsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      targetsTotal,
+			Help:      "How many Target Systems have been defined",
+		})
+
+	storesGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      storesTotal,
+			Help:      "How many Cloud Storage Systems have been defined",
+		})
+
+	jobsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      jobsTotal,
+			Help:      "How many backup jobs have been defined",
+		})
+
+	tasksGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      tasksTotal,
+			Help:      "How many tasks have been created",
+		})
+
+	archivesGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      archivesTotal,
+			Help:      "How many Backup Archives have been generated",
+		})
+
+	storageUsedBytesGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      storageBytesTotal,
+			Help:      "How much storage has been used, in bytes.",
+		})
+
+	/* To be implemented
+	targetHealthGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      targetHealthStatus,
+			Help:      "A generic health status for targets, that is differentiated into different contexts based on labels. For example, the label target=x tenant=y means that the metric applies to the health of a given target, owned by a tenant.",
+		}, []string{"tenant_name", "tenant_uuid", "target_name", "target_uuid"})
+
+	storeHealthGaugeVec = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      storeHealthStatus,
+			Help:      "A generic health status for stores, that is differentiated into different contexts based on labels. For example, the label store=x tenant=y means that the metric applies to the health of a given store, owned by a tenant.",
+		}, []string{"tenant_name", "tenant_uuid", "store_name", "store_uuid"})
+
+	coreStatusGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Name:      coreStatus,
+			Help:      "The global initialized / locked status of the SHIELD core; (0=uninitialized, 1=locked, 2=unlocked)",
+		})
+	*/
+)
+
+func InitalizeMetrics() *Metrics {
+	metrics := &Metrics{}
+	return metrics
+}
+
+func (m *Metrics) Inform(mbus *bus.Bus) {
+	m.bus = mbus
+}
+
+func (m *Metrics) RegisterExporter(tenantCount, agentsCount, targetsCount, storesCount, jobsCount, tasksCount, archivesCount float64) {
```

Comment:
Change input type to int instead of float. Cast within function.

Reply:
You should probably change all of the args here into a single struct - when you have this many args to a function, remembering the ordering becomes infeasible. Putting it into a struct makes the caller label the inputs, and also allows them to specify them in any order.

Reply:
This entire function should just be part of InitializeMetrics... which should have its name changed to `New`